src/client.rs

```diff
@@ -9,7 +9,7 @@ use std::{
 
 use futures::{ready, Stream};
 use hyper::{
-    body::{Bytes, HttpBody},
+    body::HttpBody,
```

Comment:
The original implementation had the Decoded stream wrapping the Reconnecting stream. The outer  stream dealt with Events, which gave it the ability to monitor event ids. But the inner stream was the one that needed to know the last event for subsequent sends. This made it difficult to deal with the stream id spec. So instead I decided to keep the Reconnecting stream but have it instead rely on an EventParser (essentially flipping the dependencies).